### PR TITLE
refactor(core): the `RuntimeError` class should support more compact syntax

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit refactors the `RuntimeError` class to support a short version of providing error messages:
```
throw new RuntimeError(
  RuntimeErrorCode.INJECTOR_ALREADY_DESTROYED,
  ngDevMode && 'Injector has already been destroyed.');
```
In prod mode, the second argument becomes `false` andn this commit extends the typings to support that.

This commit also contains a couple places were the `RuntimeError` class is used to demostrate the compact form.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

reviewed-for: public-api

File: packages/core/src/errors.ts [hunk 1]
```diff
@@ -65,14 +65,33 @@ export const enum RuntimeErrorCode {
...
   }
 }
 
-/** Called to format a runtime error */
-export function formatRuntimeError<T = RuntimeErrorCode>(code: T, message: string): string {
+/**
+ * Called to format a runtime error.
+ * See additional info on the `message` argument type in the `RuntimeError` class description.
+ */
+export function formatRuntimeError<T = RuntimeErrorCode>(
```
Comment (alxhub) [inline packages/core/src/errors.ts:93]:
```suggestion
export function formatRuntimeError<T extends number = RuntimeErrorCode>(
```

This gets rid of the need for the `codeAsNumber` cast.

**Review by alxhub — APPROVED** [review]

**Review by alxhub — APPROVED** [review]
Reviewed-for: public-api

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:425458614:BASE:425458657:1643667150679:b2c7f54d).

File: packages/core/src/errors.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/core/src/errors.ts]: Awesome, thanks Alex! 👍

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit ed1732c2680e7faa736be9c2eaf3640efbc41edf.

**Pull request closed without merging**